### PR TITLE
Allow specifying the full key version.

### PR DESCRIPTION
**Pull request by dlorenc**
This means we don't need to give KMS key management IAM permissions
if you can specify the full thing.

Signed-off-by: Dan Lorenc <dlorenc@google.com>

### CODE REVIEW

**Comment by dlorenc** [comment]
cc @priyawadhwa

**Review by priyawadhwa — DISMISSED** [review]
nice!

**Comment by priyawadhwa** [comment]
looks like i need write access to this repo 😅

File: pkg/kms/gcp/gcp.go [hunk 1]
```diff
@@ -39,12 +39,13 @@ type KMS struct {
 	locationID    string
 	keyRing       string
 	key           string
+	version       string
 }
 
 var (
 	ErrKMSReference = errors.New("kms specification should be in the format gcpkms://projects/[PROJECT_ID]/locations/[LOCATION]/keyRings/[KEY_RING]/cryptoKeys/[KEY]")
```
Comment (bobcallaway) [inline pkg/kms/gcp/gcp.go:46]: do you need to update this string to note the optional versions tag?

File: pkg/kms/gcp/gcp.go [hunk 2]
```diff
@@ -57,18 +58,18 @@ func ValidReference(ref string) error {
...
-func parseReference(resourceID string) (projectID, locationID, keyRing, keyName string, err error) {
+func parseReference(resourceID string) (projectID, locationID, keyRing, keyName, version string, err error) {
 	v := re.FindStringSubmatch(resourceID)
-	if len(v) != 5 {
+	if len(v) < 5 {
 		err = errors.Errorf("invalid gcpkms format %q", resourceID)
 		return
 	}
-	projectID, locationID, keyRing, keyName = v[1], v[2], v[3], v[4]
+	projectID, locationID, keyRing, keyName, version = v[1], v[2], v[3], v[4], v[5]
```
Comment (bobcallaway) [inline pkg/kms/gcp/gcp.go:67]: if `len(v) = 5` then wouldn't `v[5]` be out of bounds?

**Review by bobcallaway — COMMENTED** [review]
two questions

File: pkg/kms/gcp/gcp.go (hunk 1 above)
Comment (dlorenc) [reply pkg/kms/gcp/gcp.go]: Good catch.

**Review by bobcallaway — APPROVED** [review]
lgtm

**Pull request merged**